### PR TITLE
logout in server

### PR DESCRIPTION
**Pull request by TsvAliaksandra**

### CODE REVIEW

File: client/app/redux/modules/auth/actions.ts [hunk 1]
```diff
@@ -98,3 +98,26 @@ export const signUp: SignUp = (registeredUser: IRegisteredUser) => (dispatch: Di
       throw  err;
     });
 };
+
+export type Logout = (refreshToken: string) => (dispatch: Dispatch) => void;
+export const logout: Logout = (refreshToken: string) => (dispatch: Dispatch) => {
+  return axiosRequest.post('/api/auth/logout', { refreshToken })
+    .then((res: axios.AxiosResponse<void>) => {
+      if (res.status === 200) {
```
Comment (CeQadrat) [inline client/app/redux/modules/auth/actions.ts]: What codes other than 200 can come with a successful request? And why should we logout only if at code 200, and not simply with a successful request?

File: client/app/redux/modules/auth/actions.ts [hunk 2]
```diff
@@ -98,3 +98,26 @@ export const signUp: SignUp = (registeredUser: IRegisteredUser) => (dispatch: Di
...
+        localStorage.clear();
+
+        dispatch({
+          type: AUTH_CONSTANTS.LOGOUT,
+        });
+      }
+    })
+    .catch((err: axios.AxiosError) => {
```
Comment (Rezdoom) [inline client/app/redux/modules/auth/actions.ts]: If session is not found, server response is returned with code 401 (Unauthorized) and so you need to logout.

Comment (Rezdoom) [reply client/app/redux/modules/auth/actions.ts]: Except for error with 500 code, if 500 you should redirect to 'internal server error' page.

File: server/API/controllers/AuthController.ts [hunk 3]
```diff
@@ -64,4 +64,17 @@ export class AuthController implements interfaces.Controller {
...
   }
+
+  @httpPost('/logout')
+  private async logout(
+    @response() res: express.Response,
+    @requestBody('refreshToken') refreshToken: string,
+    @nextFn() next: express.NextFunction,
+  ): Promise<void> {
+    try {
+      res.status(200).json(await this._authService.logout(refreshToken));
```
Comment (Rezdoom) [inline server/API/controllers/AuthController.ts]: Default response code is 200

**Pull request merged**